### DISCUSSION TITLE
Sympy physics mechanics, second derivatives

### DISCUSSION THREAD

**Discussion by Peter230655** [sympy/sympy#21809] [Q&A]
In a model I am playing around with, the force on a body depends on the second derivative of a generalized coordinate.
Say, q = generalized coordinate
        u = q.diff(t)
When I put it into the force list, like FL = [(Dmc, - k0*u.diff(t)*N.x)] 
it seems that this force gets ignored, or set to zero when setting up the equations of motion. u.diff(t) was not in the dynamic symbols of the force vector, and also the result looked like it was ignored.

Then I created F =  dynamicsymbol(‚F‘), and put this into the force list, like FL = [(Dmc, - k0*F*N.x)]
Now F appeared in the force vector, I substituted like force = force.subs({u.diff(t): rhs(..)})
Now all seemed to work fine.

I wonder what I did wrong, that my first attempt did not work.
Thanks for any help!

**Comment by moorepants** [comment]
There may be a bug. I don't think we have any prior test cases of forces that are functions of accelerations.

**Comment by Peter230655** [comment]
Thanks!
My 'substitution' seemed to work, so I guess it was 'legal' in the frame
work of sympy.physics.mechanics?

Am Di., 3. Aug. 2021 um 14:00 Uhr schrieb Jason K. Moore <
***@***.***>:

**Reply by moorepants** [reply]
Yes, substituting should be fine. But if you need the mass matrix form, you'll have to form that manually to make sure you capture the coefficients.

**Comment by Peter230655** [comment]
Thanks!
So, the safe way is to use the ,rhs' to do a substitution?

On Tue 3. Aug 2021 at 15:31, Jason K. Moore ***@***.***>
wrote:

> Yes, substituting should be fine. But if you need the mass matrix form,
> you'll have to form that manually to make sure you capture the coefficients.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sympy/sympy/discussions/21809#discussioncomment-1125124>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT5MQURUP62QMDFRGNT4E63T27VTVANCNFSM5BLANDPQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>
-- 
Best regards,

Peter Stahlecker
